Firestore/core/src/firebase/firestore/model/field_value.cc

```diff
@@ -193,6 +221,13 @@ class TimestampValue : public BaseValue {
     return util::ToString(value_);
   }
 
+  bool Equals(const BaseValue& other) const override {
```

Comment:
Note that in the Timestamp case specifically, you don't need to specialize util::Hash, since std::hash is defined for Timestamp. That does apply to all the other cases though.

Regardless, deferring until absl::Hash sgtm.